### DISCUSSION TITLE
Anyone using the Mage_Persistent module?

### DISCUSSION THREAD

**Discussion by luigifab** [OpenMage/magento-lts#2025] [Ideas]
I [removed it](https://github.com/luigifab/magento-lts/commit/e6359ce7456ea269d401999c21ddd11637d9aa05) from our website. So same question as https://github.com/OpenMage/magento-lts/discussions/2024 for Mage_Backup.
I'm wondering if anyone actually finds it useful or is it a dead module that can potentially be removed in the future like Compiler and others?

**Comment by addison74** [comment]
In the stores I manage all the themes have customized versions of phtml files in folders with persistent name. I would not rush to remove it from OpenMage, being a different situation from Mage_Backup, although here too I found discussions on forums that they still use it for backup.

Although the many discussions invoke novelty in OpenMage 20 there are PR's which unfortunately were included only in this version and not in version 19. If I were to return to version 19 I would have to graft many changes that add real features.

**Reply by luigifab** [reply]
But are you using the persistent functionality in your websites?
If not, I remember that I only moved phtml outside persistent directory.

**Reply by addison74** [reply]
I am using persistent feature in my stores because it keeps track of unpurchased items which are left in the cart. Not all visitors are registered customers. Recently in an online store I was pleasantly surprised to find my shopping cart with all products there after almost 2 weeks and I revised and placed the order.

Although at first glance we may want to clean up OpenMage code it is good to analyze in all aspects when doing so. For this feature I choose to keep it in OpenMage.

Please note Persistent Shopping Cart feature is available in Magento 2 too.

**Comment by justinbeaty** [comment]
Maybe off topic to this direct discussion, but as far as project cleanup, I would really love it if we could break out some of the core modules into separate composer repos, and have the releases archive just bundle them all for legacy users.

To do that, I think we'd have to:

- [ ] Fix the Mage_Adminhtml module, see: https://github.com/OpenMage/magento-lts/issues/416#issuecomment-356348346
- [ ] Carefully remove certain modules into their own repo

I know this is probably a big task, but it could really set us free from debates on which modules we can or cannot remove.
